hoomd/simulation.py

```diff
@@ -246,7 +246,12 @@ def create_state_from_snapshot(self,
             # snapshot is hoomd.Snapshot
             self._state = State(self, snapshot, domain_decomposition)
         elif _match_class_path(snapshot, 'gsd.hoomd.Snapshot'):
-            # snapshot is gsd.hoomd.Snapshot
+            # snapshot is gsd.hoomd.Snapshot (gsd 2.x)
+            snapshot = Snapshot.from_gsd_snapshot(snapshot,
+                                                  self._device.communicator)
```

Comment:
I suppose the name of this factory method will have to be a little awkward for a while. We could change it to `from_gsd_frame` once hoomd supports only gsd>=2.8.0. Thoughts?

Reply:
Good point. Should we start requiring gsd 2.8.0 in HOOMD-blue v4? Or delay until v5?

For the latter, we can add `from_gsd_frame` now and deprecate `from_gsd_snapshot` for future removal in v5.

Reply:
I like adding `from_gsd_frame` and deprecating `from_gsd_snapshot` in v4, removing `from_gsd_snapshot` and requiring `gsd>=2.8.0` in v5. Depending on when a v5 is released, we could require `gsd>=3.0.0` in v5.